app/ante/eth.go

```diff
@@ -408,69 +399,7 @@ func (ctd CanTransferDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate
 		}
 	}
 
-	ctd.evmKeeper.WithContext(ctx)
-
-	// set the original gas meter
-	return next(ctx, tx, simulate)
-}
-
-// AccessListDecorator prepare an access list for the sender if Yolov3/Berlin/EIPs 2929 and 2930 are
-// applicable at the current block number.
-type AccessListDecorator struct {
```

Comment:
moved to x/evm/statedb/statedb.go ?

Reply:
It's not been used for a while, the logic has been moved into `ApplyMessage`.

Reply:
this is run in ante handler, not deliver tx
applymessage is run in deliver-tx.

Reply:
> this is run in ante handler, not deliver tx
> applymessage is run in deliver-tx.

ante handler runs in checkTx and deliverTx, and `ApplyMessage` run in deliverTx and query handler, that's why `ApplyMessage` is a better place for it.

Reply:
yes, so checkTx, this code was not called any more,
so previously, if some txs are rejected, but in this pr,
it can be included via delivertx

Reply:
> yes, so checkTx, this code was not called any more,
> so previously, if some txs are rejected, but in this pr,
> it can be included via delivertx

I made a separated PR for it: https://github.com/tharsis/ethermint/pull/797